src/Utils/request/useTanStackQueryInstead.ts

```diff
@@ -0,0 +1,52 @@
+import { useQuery } from "@tanstack/react-query";
+
+import request from "@/Utils/request/request";
+import {
+  QueryRoute,
+  RequestOptions,
+  RequestResult,
+} from "@/Utils/request/types";
+
+import { mergeRequestOptions } from "./utils";
+
+export interface QueryOptions<TData> extends RequestOptions<TData> {
+  prefetch?: boolean;
+  refetchOnWindowFocus?: boolean;
+  key?: string;
+}
+
+export default function useTanStackQueryInstead<TData>(
+  route: QueryRoute<TData>,
+  options?: QueryOptions<TData>,
+) {
+  const { data: response, isLoading } = useQuery({
+    queryKey: [route.path, options?.pathParams, options?.query],
+    queryFn: async () => {
+      return await request(route, options);
+    },
+    enabled: options?.prefetch ?? true,
+    refetchOnWindowFocus: options?.refetchOnWindowFocus,
+  });
+
+  /**
+   * Refetch function that applies new options and fetches fresh data.
+   */
+  const refetch = async (
+    overrides?: QueryOptions<TData>,
+  ): Promise<RequestResult<TData>> => {
+    const resolvedOptions = overrides
+      ? mergeRequestOptions(options || {}, overrides)
+      : options;
+
+    // Directly fetch data with resolved options
+    return await request(route, resolvedOptions);
+  };
```

Comment:
When calling `refetch` in this implementation, we are bypassing tanstack's useQuery invocation made above altogether. This means the returned `data`, `res`, etc. by `useTanStackQueryInstead` is only of the first request and not of subsequent requests.

Hence the `refetch` called for the current auth user when signIn is invoked, the original states aren't updated. Same for signOut and other similar usages in CARE.

Reply:
Let's instead use tanstack's useQuery's `refetch` itself instead?